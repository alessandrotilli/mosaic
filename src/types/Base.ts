export interface IBase {
  dataCy?: string;
}

export interface ILoadable extends IBase {
  loading?: boolean;
}

// TODO: door open to allow explicit property localization
export interface ILocalizable extends IBase {
  localized?: boolean;
}
