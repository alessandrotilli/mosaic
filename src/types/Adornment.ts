import { IBase } from "./Base";

type IBadgeColor = "default" | "error" | "primary" | "secondary";

type IBadgeVariant = "dot" | "standard";

export interface IBadgeAdornment {
  color?: IBadgeColor;
  value: string;
  variant?: IBadgeVariant;
}

export interface IAdornment extends IBase {
  /**
   * Badge properties
   */
  badge?: IBadgeAdornment;
  /**
   * Text of tooltip
   */
  tooltip?: string;
}
