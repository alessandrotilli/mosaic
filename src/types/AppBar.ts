import { ILocalizable } from "./Base";
import { IBaseButton } from "./Button";
import { IBaseIconButton } from "./IconButton";

export interface IAppBar extends ILocalizable {
  actions?: IBaseIconButton[];
  menu?: IBaseIconButton;
  onTitleClick?: () => void;
  title?: string;
  userMenu?: IBaseButton[];
  username?: string;
}
