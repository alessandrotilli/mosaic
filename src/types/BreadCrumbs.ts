import { IBase } from "./Base";

export type BreadCrumbsType = IBase & {
  link: { label: string; href: string }[];
  onClick: (href: string) => void;
};
