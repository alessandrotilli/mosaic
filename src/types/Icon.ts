import { CSSProperties } from "react";

import { ILoadable } from "./Base";

export enum Icons {
  account = "account",
  add = "add",
  apps = "apps",
  block = "block",
  brightness = "brightness",
  business = "business",
  chart_timeline = "chart_timeline",
  check = "check",
  checkbox = "checkbox",
  checkbox_empty = "checkbox_empty",
  clock = "clock",
  close = "close",
  connection_slow = "connection_slow",
  crop = "crop",
  delete = "delete",
  details = "details",
  down = "down",
  edit = "edit",
  error = "error",
  filter = "filter",
  first = "first",
  group = "group",
  home = "home",
  language = "language",
  last = "last",
  left = "left",
  list = "list",
  list_check = "list_check",
  mail = "mail",
  mail_open = "mail_open",
  menu = "menu",
  notifications = "notifications",
  open_new = "open_new",
  pause = "pause",
  play = "play",
  power = "power",
  print = "print",
  refresh = "refresh",
  right = "right",
  save = "save",
  search = "search",
  send = "send",
  settings = "settings",
  up = "up",
  up_arrow = "up_arrow",
  visibility = "visibility",
  visibility_off = "visibility_off",
  warn = "warn",
  wifi_on = "wifi",
  wifi_off = "wifi_offline",
}

export enum IconSize {
  default = "default",
  large = "large",
  small = "small",
}

interface IForwardedIcon {
  className?: string;
  color?: string;
  ref?: any;
}

export interface IIcon extends ILoadable {
  forwarded?: IForwardedIcon;
  name?: Icons;
  rotate?: boolean;
  size?: IconSize;
}

export type IIconDimensions = {
  [key in IconSize]: number;
};

export interface IIconUtilizer {
  icon: Icons;
}

export type IPartialIconUtilizer = Partial<IIconUtilizer>;

export interface IRenderedIcon extends IForwardedIcon {
  "data-cy": string;
  fontSize: IconSize;
  style?: CSSProperties;
}
