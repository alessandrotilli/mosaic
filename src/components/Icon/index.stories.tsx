import React from "react";
import MUIStyleIcon from "@mui/icons-material/Style";
import { ComponentMeta, ComponentStory } from "@storybook/react";

import { Icons } from "../../types/Icon";
import getDocsPage from "../../utils/stories";

import Icon, { DATA_CY_DEFAULT } from ".";

export default {
  title: "Display/Icon",
  component: Icon,
  parameters: {
    docs: {
      ...getDocsPage({
        basedOn: {
          label: "MUI Icon Component",
          url: "https://v4.mui.com/components/icons/",
        },
        component: "Icon",
        e2eTestInfo: {
          dataCyDefault: DATA_CY_DEFAULT,
        },
      }),
    },
  },
} as ComponentMeta<typeof Icon>;

const Template: ComponentStory<typeof Icon> = (args) => <Icon {...args} dataCy={DATA_CY_DEFAULT} />;

export const Primary = Template.bind({});
Primary.args = {
  name: Icons.account,
};

export const IconCustom = Template.bind({});
IconCustom.args = {
  children: <MUIStyleIcon />,
};

export const Loading = Template.bind({});
Loading.args = {
  ...Primary.args,
  loading: true,
};

export const Rotate = Template.bind({});
Rotate.args = {
  ...Primary.args,
  rotate: true,
};

export const SizeSmall = Template.bind({});
SizeSmall.args = {
  ...Primary.args,
  size: "small",
};

export const SizeLarge = Template.bind({});
SizeLarge.args = {
  ...Primary.args,
  size: "large",
};

export const Styled = Template.bind({});
Styled.args = {
  ...Primary.args,
  style: {
    backgroundColor: "red",
    borderRadius: "4px",
    color: "white",
    padding: "4px",
  },
};

// TODO: add this story using union type
// const allIcons = Object.values(Icons);
// const getIcon = (icon: string) => allIcons.find((i) => i === icon);
// const allIconsJsx = allIcons.map((icon, index) => (
//   <div key={index} className="icon-wrapper">
//     <Icon dataCy={`icon-${icon}`} name={getIcon(icon) || Icons.add} />
//     <span>{icon}</span>
//   </div>
// ));

// export const AllIcons = () => <StoriesWrapper>{allIconsJsx}</StoriesWrapper>;
