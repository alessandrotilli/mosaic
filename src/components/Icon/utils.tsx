import React from "react";
import MUIAccountCircleIcon from "@material-ui/icons/AccountCircle";
import MUIAddIcon from "@material-ui/icons/Add";
import MUIAppsIcon from "@material-ui/icons/Apps";
import MUIArrowUpwardIcon from "@material-ui/icons/ArrowUpward";
import MUIBlockIcon from "@material-ui/icons/Block";
import MUIBrightness4Icon from "@material-ui/icons/Brightness4";
import MUIBusinessIcon from "@material-ui/icons/Business";
import MUICheckIcon from "@material-ui/icons/Check";
import MUICheckBoxIcon from "@material-ui/icons/CheckBox";
import MUICheckBoxOutlineBlankIcon from "@material-ui/icons/CheckBoxOutlineBlank";
import MUIChevronLeftIcon from "@material-ui/icons/ChevronLeft";
import MUIChevronRightIcon from "@material-ui/icons/ChevronRight";
import MUICloseIcon from "@material-ui/icons/Close";
import MUICropIcon from "@material-ui/icons/Crop";
import MUIDeleteIcon from "@material-ui/icons/Delete";
import MUIDetailsIcon from "@material-ui/icons/Details";
import MUIDraftsIcon from "@material-ui/icons/Drafts";
import MUIEditIcon from "@material-ui/icons/Edit";
import MUIErrorIcon from "@material-ui/icons/Error";
import MUIFilterListIcon from "@material-ui/icons/FilterList";
import MUIFirstPageIcon from "@material-ui/icons/FirstPage";
import MUIGroupWorkIcon from "@material-ui/icons/GroupWork";
import MUIHomeIcon from "@material-ui/icons/Home";
import MUIKeyboardArrowDownIcon from "@material-ui/icons/KeyboardArrowDown";
import MUIKeyboardArrowUpIcon from "@material-ui/icons/KeyboardArrowUp";
import MUILanguageIcon from "@material-ui/icons/Language";
import MUILastPageIcon from "@material-ui/icons/LastPage";
import MUIListIcon from "@material-ui/icons/List";
import MUIMailIcon from "@material-ui/icons/Mail";
import MUIMenuIcon from "@material-ui/icons/Menu";
import MUINotificationsIcon from "@material-ui/icons/Notifications";
import MUIOpenInNew from "@material-ui/icons/OpenInNew";
import MUIPauseIcon from "@material-ui/icons/Pause";
import MUIPlayArrowIcon from "@material-ui/icons/PlayArrow";
import MUIPlaylistAddCheckIcon from "@material-ui/icons/PlaylistAddCheck";
import MUIPowerSettingsNewIcon from "@material-ui/icons/PowerSettingsNew";
import MUIPrintIcon from "@material-ui/icons/Print";
import MUIRefreshIcon from "@material-ui/icons/Refresh";
import MUISaveIcon from "@material-ui/icons/Save";
import MUIScheduleIcon from "@material-ui/icons/Schedule";
import MUISearchIcon from "@material-ui/icons/Search";
import MUISendIcon from "@material-ui/icons/Send";
import MUISettingsIcon from "@material-ui/icons/Settings";
import MUISignalWifi1BarOutlinedIcon from "@material-ui/icons/SignalWifi1BarOutlined";
import MUITimelineIcon from "@material-ui/icons/Timeline";
import MUIVisibilityIcon from "@material-ui/icons/Visibility";
import MUIVisibilityOffIcon from "@material-ui/icons/VisibilityOff";
import MUIWarningIcon from "@material-ui/icons/Warning";
import MUIWifiOffOutlinedIcon from "@material-ui/icons/WifiOffOutlined";
import MUIWifiOutlinedIcon from "@material-ui/icons/WifiOutlined";

import { Icons, IIconsCatalog } from "../../types/Icon";

export const iconsCatalog: IIconsCatalog = {
  [Icons.account]: <MUIAccountCircleIcon />,
  [Icons.add]: <MUIAddIcon />,
  [Icons.apps]: <MUIAppsIcon />,
  [Icons.block]: <MUIBlockIcon />,
  [Icons.brightness]: <MUIBrightness4Icon />,
  [Icons.business]: <MUIBusinessIcon />,
  [Icons.chart_timeline]: <MUITimelineIcon />,
  [Icons.check]: <MUICheckIcon />,
  [Icons.checkbox]: <MUICheckBoxIcon />,
  [Icons.checkbox_empty]: <MUICheckBoxOutlineBlankIcon />,
  [Icons.clock]: <MUIScheduleIcon />,
  [Icons.close]: <MUICloseIcon />,
  [Icons.connection_slow]: <MUISignalWifi1BarOutlinedIcon />,
  [Icons.crop]: <MUICropIcon />,
  [Icons.delete]: <MUIDeleteIcon />,
  [Icons.details]: <MUIDetailsIcon />,
  [Icons.down]: <MUIKeyboardArrowDownIcon />,
  [Icons.edit]: <MUIEditIcon />,
  [Icons.error]: <MUIErrorIcon />,
  [Icons.filter]: <MUIFilterListIcon />,
  [Icons.first]: <MUIFirstPageIcon />,
  [Icons.group]: <MUIGroupWorkIcon />,
  [Icons.home]: <MUIHomeIcon />,
  [Icons.language]: <MUILanguageIcon />,
  [Icons.last]: <MUILastPageIcon />,
  [Icons.left]: <MUIChevronLeftIcon />,
  [Icons.list_check]: <MUIPlaylistAddCheckIcon />,
  [Icons.list]: <MUIListIcon />,
  [Icons.mail_open]: <MUIDraftsIcon />,
  [Icons.mail]: <MUIMailIcon />,
  [Icons.menu]: <MUIMenuIcon />,
  [Icons.notifications]: <MUINotificationsIcon />,
  [Icons.open_new]: <MUIOpenInNew />,
  [Icons.pause]: <MUIPauseIcon />,
  [Icons.play]: <MUIPlayArrowIcon />,
  [Icons.power]: <MUIPowerSettingsNewIcon />,
  [Icons.print]: <MUIPrintIcon />,
  [Icons.refresh]: <MUIRefreshIcon />,
  [Icons.right]: <MUIChevronRightIcon />,
  [Icons.save]: <MUISaveIcon />,
  [Icons.search]: <MUISearchIcon />,
  [Icons.send]: <MUISendIcon />,
  [Icons.settings]: <MUISettingsIcon />,
  [Icons.up_arrow]: <MUIArrowUpwardIcon />,
  [Icons.up]: <MUIKeyboardArrowUpIcon />,
  [Icons.visibility]: <MUIVisibilityIcon />,
  [Icons.visibility_off]: <MUIVisibilityOffIcon />,
  [Icons.warn]: <MUIWarningIcon />,
  [Icons.wifi_off]: <MUIWifiOffOutlinedIcon />,
  [Icons.wifi_on]: <MUIWifiOutlinedIcon />,
};
