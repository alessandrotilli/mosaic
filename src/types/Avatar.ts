import { ILoadable } from "./Base";
import { Icons } from "./Icon";

export enum AvatarVariant {
  default = "circular",
  rounded = "rounded",
  squared = "square",
}

export interface IAvatar extends ILoadable {
  alt?: string;
  icon?: Icons;
  src?: string;
  text?: string;
  variant?: AvatarVariant;
}
