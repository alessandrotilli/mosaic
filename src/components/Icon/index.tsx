import React, { cloneElement, FC, PropsWithChildren, ReactElement, useCallback, useMemo } from "react";
import { ClassNames } from "@emotion/react";
import { keyframes, Skeleton as MUISkeleton } from "@mui/material";

import { IIcon, IIconDimensions, IRenderedIcon } from "../../types/Icon";
import { logWarn } from "../../utils/logger";
import Adornment from "../Adornment";

import { iconsCatalog } from "./utils";

export const DATA_CY_DEFAULT = "icon";

const ICON_DIMENSIONS: IIconDimensions = {
  small: 20,
  medium: 24,
  large: 35,
};

const Icon: FC<PropsWithChildren<IIcon>> = ({
  badge,
  children,
  dataCy = DATA_CY_DEFAULT,
  forwarded = {},
  loading = false,
  name,
  rotate = false,
  size = "medium",
  style: externalStyle,
  tooltip,
}) => {
  const dimensions = useMemo(() => {
    const dimension = ICON_DIMENSIONS[size];
    return { height: dimension, width: dimension };
  }, [size]);

  const props = useMemo(() => {
    let props: IRenderedIcon = {
      ...forwarded,
      "data-cy": dataCy,
      fontSize: size,
    };

    if (externalStyle) {
      props = { ...props, style: { ...externalStyle } };
    }

    if (children) {
      props = { ...props, style: { ...props.style, ...dimensions } };
    }

    return props;
  }, [children, dataCy, dimensions, externalStyle, forwarded, size]);

  const rotateKeyframes = useMemo(
    () => keyframes`
      from: {
        transform: "rotate(0deg)";
      }
      to: {
        transform: "rotate(360deg)";
      }
  `,
    []
  );

  const renderWithAnimation = useCallback(
    (element: any) => {
      return (
        <ClassNames>
          {({ css }) => {
            let animation = "none";
            if (rotate) {
              animation = css`
                ${rotateKeyframes} 2s linear infinite
              `;
            }

            let className = css`
              animation: ${animation};
            `;

            if (forwarded.className) {
              className = forwarded.className;
            }

            return (
              <Adornment badge={badge} tooltip={tooltip}>
                {cloneElement(element as ReactElement<any>, { ...props, className })}
              </Adornment>
            );
          }}
        </ClassNames>
      );
    },
    [badge, forwarded, props, rotate, rotateKeyframes, tooltip]
  );

  if (loading) {
    return <MUISkeleton variant="rectangular" style={{ ...dimensions }} />;
  }

  if (children) {
    return renderWithAnimation(children);
  }

  if (!name) {
    logWarn("Icon", "Skip rendering, both children and name are not set");
    return null;
  }

  const icon = iconsCatalog[name];
  return renderWithAnimation(icon);
};

export default Icon;
