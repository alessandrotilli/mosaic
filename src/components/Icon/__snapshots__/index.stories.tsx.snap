// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display/Icon Badge smoke-test 1`] = `
<span data-cy="adornment">
  <span data-cy="adornment-badge"
        class="MuiBadge-root css-mui-classname"
  >
    <svg class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-mui-classname"
         focusable="false"
         aria-hidden="true"
         viewbox="0 0 24 24"
         data-testid="AccountCircleIcon"
         data-cy="icon"
    >
      <path d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2m0 4c1.93 0 3.5 1.57 3.5 3.5S13.93 13 12 13s-3.5-1.57-3.5-3.5S10.07 6 12 6m0 14c-2.03 0-4.43-.82-6.14-2.88C7.55 15.8 9.68 15 12 15s4.45.8 6.14 2.12C16.43 19.18 14.03 20 12 20">
      </path>
    </svg>
    <span class="MuiBadge-badge MuiBadge-standard MuiBadge-anchorOriginTopRight MuiBadge-anchorOriginTopRightRectangular MuiBadge-overlapRectangular MuiBadge-colorSecondary css-mui-classname">
      5
    </span>
  </span>
</span>
`;

exports[`Display/Icon ForwardedProps smoke-test 1`] = `
<svg class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium test-forward css-mui-classname"
     focusable="false"
     aria-hidden="true"
     viewbox="0 0 24 24"
     data-testid="AccountCircleIcon"
     data-cy="icon"
>
  <path d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2m0 4c1.93 0 3.5 1.57 3.5 3.5S13.93 13 12 13s-3.5-1.57-3.5-3.5S10.07 6 12 6m0 14c-2.03 0-4.43-.82-6.14-2.88C7.55 15.8 9.68 15 12 15s4.45.8 6.14 2.12C16.43 19.18 14.03 20 12 20">
  </path>
</svg>
`;

exports[`Display/Icon IconCustom smoke-test 1`] = `
<svg class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-mui-classname"
     focusable="false"
     aria-hidden="true"
     viewbox="0 0 24 24"
     data-testid="StyleIcon"
     data-cy="icon"
     style="height: 24px; width: 24px;"
>
  <path d="m2.53 19.65 1.34.56v-9.03l-2.43 5.86c-.41 1.02.08 2.19 1.09 2.61m19.5-3.7L17.07 3.98c-.31-.75-1.04-1.21-1.81-1.23-.26 0-.53.04-.79.15L7.1 5.95c-.75.31-1.21 1.03-1.23 1.8-.01.27.04.54.15.8l4.96 11.97c.31.76 1.05 1.22 1.83 1.23.26 0 .52-.05.77-.15l7.36-3.05c1.02-.42 1.51-1.59 1.09-2.6M7.88 8.75c-.55 0-1-.45-1-1s.45-1 1-1 1 .45 1 1-.45 1-1 1m-2 11c0 1.1.9 2 2 2h1.45l-3.45-8.34z">
  </path>
</svg>
`;

exports[`Display/Icon Loading smoke-test 1`] = `
<span class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-mui-classname"
      style="width: 24px; height: 24px;"
>
</span>
`;

exports[`Display/Icon Primary smoke-test 1`] = `
<svg class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-mui-classname"
     focusable="false"
     aria-hidden="true"
     viewbox="0 0 24 24"
     data-testid="AccountCircleIcon"
     data-cy="icon"
>
  <path d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2m0 4c1.93 0 3.5 1.57 3.5 3.5S13.93 13 12 13s-3.5-1.57-3.5-3.5S10.07 6 12 6m0 14c-2.03 0-4.43-.82-6.14-2.88C7.55 15.8 9.68 15 12 15s4.45.8 6.14 2.12C16.43 19.18 14.03 20 12 20">
  </path>
</svg>
`;

exports[`Display/Icon Rotate smoke-test 1`] = `
<svg class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-mui-classname"
     focusable="false"
     aria-hidden="true"
     viewbox="0 0 24 24"
     data-testid="AccountCircleIcon"
     data-cy="icon"
>
  <path d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2m0 4c1.93 0 3.5 1.57 3.5 3.5S13.93 13 12 13s-3.5-1.57-3.5-3.5S10.07 6 12 6m0 14c-2.03 0-4.43-.82-6.14-2.88C7.55 15.8 9.68 15 12 15s4.45.8 6.14 2.12C16.43 19.18 14.03 20 12 20">
  </path>
</svg>
`;

exports[`Display/Icon SizeLarge smoke-test 1`] = `
<svg class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge css-mui-classname"
     focusable="false"
     aria-hidden="true"
     viewbox="0 0 24 24"
     data-testid="AccountCircleIcon"
     data-cy="icon"
>
  <path d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2m0 4c1.93 0 3.5 1.57 3.5 3.5S13.93 13 12 13s-3.5-1.57-3.5-3.5S10.07 6 12 6m0 14c-2.03 0-4.43-.82-6.14-2.88C7.55 15.8 9.68 15 12 15s4.45.8 6.14 2.12C16.43 19.18 14.03 20 12 20">
  </path>
</svg>
`;

exports[`Display/Icon SizeSmall smoke-test 1`] = `
<svg class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-mui-classname"
     focusable="false"
     aria-hidden="true"
     viewbox="0 0 24 24"
     data-testid="AccountCircleIcon"
     data-cy="icon"
>
  <path d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2m0 4c1.93 0 3.5 1.57 3.5 3.5S13.93 13 12 13s-3.5-1.57-3.5-3.5S10.07 6 12 6m0 14c-2.03 0-4.43-.82-6.14-2.88C7.55 15.8 9.68 15 12 15s4.45.8 6.14 2.12C16.43 19.18 14.03 20 12 20">
  </path>
</svg>
`;

exports[`Display/Icon SkipRender smoke-test 1`] = `""`;

exports[`Display/Icon Styled smoke-test 1`] = `
<svg class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-mui-classname"
     focusable="false"
     aria-hidden="true"
     viewbox="0 0 24 24"
     data-testid="AccountCircleIcon"
     data-cy="icon"
     style="background-color: red; border-radius: 4px; color: white; padding: 4px;"
>
  <path d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2m0 4c1.93 0 3.5 1.57 3.5 3.5S13.93 13 12 13s-3.5-1.57-3.5-3.5S10.07 6 12 6m0 14c-2.03 0-4.43-.82-6.14-2.88C7.55 15.8 9.68 15 12 15s4.45.8 6.14 2.12C16.43 19.18 14.03 20 12 20">
  </path>
</svg>
`;

exports[`Display/Icon Tooltip smoke-test 1`] = `
<span data-cy="adornment">
  <span aria-label="account"
        data-cy="adornment-tooltip"
        class
  >
    <svg class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-mui-classname"
         focusable="false"
         aria-hidden="true"
         viewbox="0 0 24 24"
         data-testid="AccountCircleIcon"
         data-cy="icon"
    >
      <path d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2m0 4c1.93 0 3.5 1.57 3.5 3.5S13.93 13 12 13s-3.5-1.57-3.5-3.5S10.07 6 12 6m0 14c-2.03 0-4.43-.82-6.14-2.88C7.55 15.8 9.68 15 12 15s4.45.8 6.14 2.12C16.43 19.18 14.03 20 12 20">
      </path>
    </svg>
  </span>
</span>
`;
